concrete/src/Page/Sitemap/SitemapGenerator.php

```diff
@@ -0,0 +1,343 @@
+<?php
+
+namespace Concrete\Core\Page\Sitemap;
+
+use Concrete\Core\Application\Application;
+use Concrete\Core\Attribute\Category\PageCategory;
+use Concrete\Core\Cache\Cache;
+use Concrete\Core\Page\Page;
+use Concrete\Core\Page\Sitemap\Element\SitemapFooter;
+use Concrete\Core\Page\Sitemap\Element\SitemapHeader;
+use Concrete\Core\Page\Sitemap\Element\SitemapPage;
+use Concrete\Core\Page\Sitemap\Element\SitemapPageAlternativeLanguage;
+use Concrete\Core\Url\Resolver\Manager\ResolverManagerInterface;
+use DateTime;
+
+/**
+ * Class to be used to generate the elements to be included in a sitemap.xml file.
+ */
+class SitemapGenerator
+{
+    /**
+     * @var \Concrete\Core\Application\Application
+     */
+    protected $app;
+
+    /**
+     * @var \Concrete\Core\Page\Sitemap\PageListGenerator
+     */
+    protected $pageListGenerator;
+
+    /**
+     * @var \Concrete\Core\Config\Repository\Repository
+     */
+    protected $config;
+
+    /**
+     * @var string
+     */
+    protected $customSiteCanonicalUrl = '';
+
+    /**
+     * @var \Concrete\Core\Url\Resolver\Manager\ResolverManagerInterface
+     */
+    private $resolverManager;
+
+    /**
+     * @var \Concrete\Core\Entity\Attribute\Key\PageKey|null|false
+     */
+    private $sitemapChangeFrequencyAttributeKey = false;
+
+    /**
+     * @var \Concrete\Core\Entity\Attribute\Key\PageKey|null|false
+     */
+    private $sitemapPriorityAttributeKey = false;
+
+    /**
+     * @var string|false
+     */
+    private $defaultChangeFrequency = false;
+
+    /**
+     * @var string|false
+     */
+    private $defaultPriority = false;
+
+    /**
+     * Initialize the instance.
+     *
+     * @param \Concrete\Core\Application\Application $app
+     * @param \Concrete\Core\Page\Sitemap\PageListGenerator $pageListGenerator
+     * @param \Concrete\Core\Url\Resolver\Manager\ResolverManagerInterface $resolverManager
+     */
+    public function __construct(Application $app, PageListGenerator $pageListGenerator, ResolverManagerInterface $resolverManager)
+    {
+        $this->app = $app;
+        $this->pageListGenerator = $pageListGenerator;
+        $this->resolverManager = $resolverManager;
+        $this->config = $this->app->make('config');
```

Comment:
Is there a reason why you don't DI the `Repository` class?

Reply:
Not at all :wink: